File: part1-coding/02-sliding-window-longest-substring/java/02SlidingWindowLongestSubstring.java
import java.util.*;
public class LongestSubstring {
    public static int lengthOfLongestSubstring(String s){
        Set<Character> set = new HashSet<>();
        int l=0,b=0;
        for(int r=0;r<s.length();r++){
            while(set.contains(s.charAt(r))) set.remove(s.charAt(l++));
            set.add(s.charAt(r));
            b = Math.max(b, r-l+1);
        }
        return b;
    }
}

File: part1-coding/02-sliding-window-longest-substring/python/02_sliding_window_longest_substring.py
def longest_unique(s: str) -> int:
    seen, L, best = set(), 0, 0
    for R, ch in enumerate(s):
        while ch in seen:
            seen.remove(s[L]); L += 1
        seen.add(ch)
        best = max(best, R - L + 1)
    return best
